### DISCUSSION TITLE
Look for info regarding "the major browsers do not support requests with `no-store`."

### DISCUSSION THREAD

**Discussion by curtisbelt** [mdn/mdn-community#507] [Content] [answered]
https://github.com/mdn/content/commit/2aff7c207a4ae378a947a706b0dfea0b7066c097#diff-0e775ebf01146f89262a34438d65c3660c6d7884f7423355307210753b3f90e7R284

> Note that the major browsers do not support requests with `no-store`.

This line was added 2 years ago, co-authors @Jxck @hamishwillee @mnot @sideshowbarker.

I cannot find any reference online that backs up this claim. The latest Chrome/Chromium-powered browsers support `no-store`. I was just curious if anyone has a source on this or more details.

Thank you!

**Comment by mnot** [comment]
Chrome and Safari don't support request `no-store`; Firefox does. See https://cache-tests.fyi/#cc-request

That said, the statement is odd; what there really should be is a statement at the top of the request CC section saying that request cache-control is optional, and so not consistently supported.

**Reply by curtisbelt** [reply]
Oh! I see my mistake... I didn't make the distinction between **Request** and **Response** directives.

My use case is for using the `no-store` directive in the Cache-Control **response** header, which I see from https://cache-tests.fyi/#cc-response that it's supported across the board.

Thank you for introducing me to that cache testing resource 👍🏻

**Comment by Jxck** [comment, marked as answer]
In no-cache section above that line said

> Browsers usually add no-cache to requests when users are force reloading a page.

so I think I intended to say like

> Note that the major browsers do not provide UI for sending requests with `no-store`.

basically (really basically), browsers sending 3 types of request in user journey. "Navigation", "Reload", and "Force Reload".
And the major browser basically sends "Last-Modified + ETag", "max-age=0", "no-cache" for each.

So this line doesn't intend to mention supporting "no-store" in `fetch()` or so.

**Reply by curtisbelt** [reply]
Thank you for your response -- it turns out my original mistake was not catching that I was reading under the "request" directives, when my concern was for "response" directives (which I see is supported just fine).

Marking this as the answer though, since you're the original author of that documentation and have provided additional explanation -- appreciate it!

**Reply by Jxck** [reply]
I feel like we can remove this line since there are less information. I'll create PR soon.

**Reply by Jxck** [reply]
#28478
